device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/Message.java

```diff
@@ -212,6 +246,15 @@ public Message(String body)
     
     // ----- Public Methods -----
 
+    /**
+     * Indicates whether or not this instance is a bulk message instance.
+     * @return Boolean value to indicate whether or not this is a bulk message instance.
+     */
+    public boolean isBulk()
+    {
+        return this.isBulk;
```

Comment:
What about having a class ```BulkMessage``` that extends from this message class instead? That way in the lower layers, you can have code like

```java
if (message instanceof BulkMessage)
{
// bulk message logic
}
else
{
//normal message logic
}

Reply:
That way getNestedMessages() can only be called on a bulk message

Reply:
that is the first approach I tried, If we decide to create a BulkMessage class (which I did initially) we have to expose it (which I am not a huge fan of doing). 
The reason is that this class is used in other packages device.Transport for example. so we can't make it internal (not a public/private class). So I baked it into the same Message class.

Reply:
I'm not sure I follow why making a hypothetical ```BulkMessage``` class public would be bad when it extends an already public class.

Reply:
I made a separate class and made the constructor internal.